core/src/main/java/org/infinispan/configuration/cache/TakeOfflineConfigurationBuilder.java

```diff
@@ -26,8 +26,17 @@ public TakeOfflineConfigurationBuilder(ConfigurationBuilder builder, BackupConfi
    }
 
    /**
-    * The minimal number of millis to wait before taking this site offline, even in the case 'afterFailures' is reached.
-    * If smaller or equal to 0, then only 'afterFailures' is considered.
+    * The minimal number of millis to wait before taking this site offline.
+    * <p>
+    * The timer is reset when a successful request is performed.
```

Comment:
When does the timer start though?

Reading the code it looks like there must be at least 2 communication failures before a site can be taken offline: one to start the timer, and one to detect that the timer expired. I believe if you clarified that in the javadoc, then 

a) you wouldn't need to update the status without a failure in the tests, because we shouldn't test something that can never happen in production
b)  you wouldn't have to spell out what happens in each possible combination of `minTimeToWait` and `afterFailures`, because any `afterFailures < 2` value behaves just like `afterFailures == 2`. BTW, you're missing `minTimeToWait > 0 && afterFailures <= 0` and `minTimeToWait <= 0 && afterFailures > 0`.

Reply:
> When does the timer start though?
> 
here: https://github.com/infinispan/infinispan/pull/7505/files#diff-34176490e83e76d9a484640f2ab41942R50

when the first failure happens, it records the send timestamp

> Reading the code it looks like there must be at least 2 communication failures before a site can be taken offline: one to start the timer, and one to detect that the timer expired. I believe if you clarified that in the javadoc, then
> 
I disagree. Taking 2 communication (or more) is irrelevant for the user. It is an implementation detail and it may change. 
For the user point of view, it just needs to know that the site is going to be taken offline after `minWait` milliseconds.

> a) you wouldn't need to update the status without a failure in the tests, because we shouldn't test something that can never happen in production
I don't want to do puts inside the eventually code just to get the method invoked. 

> b) you wouldn't have to spell out what happens in each possible combination of `minTimeToWait` and `afterFailures`, because any `afterFailures < 2` value behaves just like `afterFailures == 2`. 
not really. 
`afterFailures == 1` changes the site status to offline on the first failure (assuming `minTimeToWait == 0`) 

> BTW, you're missing `minTimeToWait > 0 && afterFailures <= 0` and `minTimeToWait <= 0 && afterFailures > 0`.
I did a table for the docs with all the cases:
https://github.com/infinispan/infinispan/pull/7505/files#diff-c4005752578de1581d93923f6a6acdb7R170

Reply:
> when the first failure happens, it records the send timestamp

But the javadoc doesn't mention it

> I disagree. Taking 2 communication (or more) is irrelevant for the user. It is an implementation detail and it may change.
> For the user point of view, it just needs to know that the site is going to be taken offline after `minWait` milliseconds.

The problem with this explanation is that it doesn't leave any wiggle room. If we say in the documentation "the site is going to be taken offline after `minWait` milliseconds", then we MUST take the site offline after `minWait` milliseconds, not when the next request fails after `10 * minWait` milliseconds.

> I don't want to do puts inside the eventually code just to get the method invoked.

IMO you only need to do `eventually(() -> minTimeHasElapsed())` and a single put afterwards to force the site to be taken offline.

> `afterFailures == 1` changes the site status to offline on the first failure (assuming `minTimeToWait == 0`)

> I did a table for the docs with all the cases:
> https://github.com/infinispan/infinispan/pull/7505/files#diff-c4005752578de1581d93923f6a6acdb7R170

Maybe link to that table in the javadoc?

Reply:
> then we MUST take the site offline after `minWait` milliseconds, not when the next request fails after `10 * minWait` milliseconds.

technically,  `10 * minWait` milliseconds happens after `minWait` millisecond :)